lib/rubygems/remote_fetcher.rb

```diff
@@ -284,10 +285,19 @@ def fetch_path(uri, mtime = nil, head = false)
   end
 
   def fetch_s3(uri, mtime = nil, head = false)
-    public_uri = sign_s3_url(uri)
+    begin
+      public_uri = s3_uri_signer(uri).sign
```

Comment:
Do we need `s3_uri_signer` method? Isn't it possible to use `Gem::S3URISigner.new(uri).sign` or make shortcut as a `Gem::S3URISigner.sign(uri)` here?

Reply:
It's used to override `Gem::S3URISigner` in the unit-test. Checkout `assert_fetch_s3` in `test/rubygems/test_gem_remote_fetcher.rb`:

```
def fetcher.s3_uri_signer(uri)
  s3_uri_signer = Gem::S3URISigner.new(uri)
  def s3_uri_signer.ec2_metadata
    $instance_profile
  end
  s3_uri_signer
end
```